lapis/cli/simulate.py

```diff
@@ -131,30 +201,111 @@ def static(ctx, job_file, pool_file, storage_files, remote_throughput, cache_hit
+    default=10,
+    help="Parameter to set the network bandwidth to remote",
+)
+@click.option(
+    "--filebased_caching",
+    "filebased_caching",
+    is_flag=True,
+    help="Flag to set filebased caching on/off",
+    default=False,
+)
+@click.option("--cache-hitrate", "cache_hitrate", type=float, default=None)
 @click.pass_context
-def dynamic(ctx, job_file, pool_file):
+def dynamic(
+    ctx,
+    job_file,
+    pre_job_rank,
+    machine_ads,
+    job_ads,
+    scheduler_type,
+    pool_files,
+    storage_files,
+    remote_throughput,
+    filebased_caching,
+    cache_hitrate,
+):
     click.echo("starting dynamic environment")
     simulator = Simulator(seed=ctx.obj["seed"])
-    file, file_type = job_file
+    infile, file_type = job_file
     simulator.create_job_generator(
-        job_input=file,
+        job_input=infile,
         job_reader=partial(
             job_import_mapper[file_type],
             calculation_efficiency=ctx.obj["calculation_efficiency"],
         ),
     )
-    simulator.create_scheduler(scheduler_type=CondorJobScheduler)
-    for current_pool in pool_file:
-        file, file_type = current_pool
+    click.echo(
+        "scheduler configuration: \n "
+        "\tscheduler type: {}\n\n"
+        "\tpre job rank: {} \n\n"
+        "\tmachine classads:\n \t{}\n\n"
+        "\tjob classads: {}".format(scheduler_type, pre_job_rank, machine_ads, job_ads)
+    )
+
+    if scheduler_import_mapper[scheduler_type] == CondorClassadJobScheduler and any(
+        (pre_job_rank, machine_ads, job_ads)
+    ):
+        simulator.job_scheduler = CondorClassadJobScheduler(
+            job_queue=simulator.job_queue,
+            pre_job_rank=pre_job_rank,
+            machine_ad=machine_ads,
+            job_ad=job_ads,
+        )
+    else:
+        simulator.create_scheduler(
+            scheduler_type=scheduler_import_mapper[scheduler_type]
+        )
+
+    for current_storage_files in storage_files:
+        assert all(current_storage_files), "All storage inputs have to be set"
+        simulator.create_connection_module(remote_throughput, filebased_caching)
+        storage_file, storage_content_file, storage_type = current_storage_files
+        simulator.create_storage(
+            storage_input=storage_file,
+            storage_content_input=storage_content_file,
+            storage_reader=storage_import_mapper[storage_type],
+            storage_type=FileBasedHitrateStorage,  # TODO: Generalize this
+        )
+    for current_pool in pool_files:
+        pool_file, pool_file_type = current_pool
+        if "dummycluster" in pool_file.name:
+            simulator.create_connection_module(float("Inf"))
         simulator.create_pools(
-            pool_input=file,
-            pool_reader=pool_import_mapper[file_type],
+            pool_input=pool_file,
+            pool_reader=pool_import_mapper[pool_file_type],
             pool_type=Pool,
             controller=SimulatedLinearController,
         )
```

Comment:
This seems to be duplicated code. Please try to avoid this.

Reply:
What is your suggestion? Shall I include this in the `cli` method or write a helper function?

Reply:
Try to factor out all common code to a helper function. There seems to be a lot of overlap indeed.